### PR TITLE
Convert 'viewport.js' to ES6

### PR DESCRIPTION
**Pull request by lps88**
ES6 ✨ 

First go at this (https://github.com/guardian/frontend/pull/17220) the tests failed after merging, so this is attempt two with some refactoring and passing tests ( :crossed_fingers: )

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        iframes[iframe.id] = false;
+        iframeCounter -= 1;
+    }
+
+    if (iframeCounter === 0) {
+        w.removeEventListener('resize', onResize);
+    }
+};
+
+const onMessage = (respond: any, start, iframe: ?Element): void => {
```
Comment (gidsg) [inline static/src/javascripts/projects/commercial/modules/messenger/viewport.js]: could add a type annotation for `start` parameter?

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.spec.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        }
+    });
+
+    it('should send viewport dimensions as soon as the iframe starts listening', done => {
+        const size = {
+            width: 800,
+            height: 600,
+        };
+        detect.getViewport.mockReturnValue(size);
+        // $FlowFixMe
```
Comment (gidsg) [inline static/src/javascripts/projects/commercial/modules/messenger/viewport.spec.js]: Wonder if it is worth adding a comment to explain why this is needed?

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import detect from 'lib/detect';
+import fastdom from 'lib/fastdom-promise';
+import { register } from 'commercial/modules/messenger';
+
+let w = window;
+let iframes = {};
+let iframeCounter = 0;
+let taskQueued = false;
+
+const lastViewportRead = () => fastdom.read(() => detect.getViewport());
```
Comment (gidsg) [inline static/src/javascripts/projects/commercial/modules/messenger/viewport.js]: Probably my lack of understanding, but does this value need to be updated, is it OK as a `const`?

**Review by gidsg — APPROVED** [review]
LGTM (aside from minor comments) 👍

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.js (hunk 3 above)
Comment (lps88) [inline static/src/javascripts/projects/commercial/modules/messenger/viewport.js]: the value is a function that returns the current viewport, so it's fine as a const 👍

File: static/src/javascripts/projects/commercial/modules/messenger/viewport.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+const reset = (window_: WindowProxy): void => {
+    w = window_ || window;
+    taskQueued = false;
+    iframes = {};
+    iframeCounter = 0;
+};
+
+const sendViewportDimensions = (iframeId, viewport): void => {
+    iframes[iframeId].respond(null, viewport);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/messenger/viewport.js]: can we null check this, whilst we are converting? I've seen it throw errors.

**Review by rich-nguyen — APPROVED** [review]
Last conversion! 👍

**Pull request merged**